src/MediaPresenter.ts

```diff
@@ -0,0 +1,157 @@
+import * as mobx from "mobx";
+
+const MAXLEN = 10;
+
+export class MediaStore {
+  file: File;
+  duration: number = 0;
+  played: number = 0;
+  constructor(newFile: File) {
+    this.file = newFile;
+    // duration of each media attached
+    this.duration = 0;
+    // played list shows how long each of the file have been played
+    this.played = 0;
```

Comment:
These don't need to be instantiated as you've already set their default values on L7 & L8